### ISSUE TITLE
error C2953 when building source that includes boost/gil.hpp and boost/process.hpp

### ISSUE THREAD

**Issue by mvfpoa** [boostorg/boost#673]
Windows 10
Visual Studio 2019
Boost Version 1.76.0

Create a simple source like this:
```
#include <boost/gil.hpp>
#include <boost/process.hpp>
```

invoke

`cl.exe simple.cpp /I c:\boost\`

and you are going to get the following error message:

```
.\boost/fusion/adapted/std_tuple/tag_of.hpp(38): error C2953: 'boost::mpl::sequence_tag<std::tuple<T...>>': class template already defined
.\boost/mp11/mpl_tuple.hpp(21): note: check the declaration of 'boost::mpl::sequence_tag<std::tuple<T...>>'
```

I translated the error message to English

**Comment by mvfpoa** [comment]
possible correlation with https://github.com/boostorg/mp11/issues/39

**Comment by mclow** [comment]
I just tried to reproduce this with boost trunk and failed. (apple clang, and Mac OS)

`clang++ -std=c++14 -I $BOOST bug.cpp` compiles w/o error.

**Comment by mclow** [comment]
And I *was* able to reproduce it (apple clang, Mac OS) with the 1.79.0 release.
Please try the 1.80 beta release (available here: https://boostorg.jfrog.io/artifactory/main/beta/1.80.0.beta1/source/) and report back here if you can reproduce it there.

**Comment by mvfpoa** [comment]
Confirmed the issue is fixed with the 1.80 bera1. Thanks, @mclow!
